File: backend/app/schemas/catalog.py
"""
Catalog schemas for categories and amenities filter UI.
"""

from pydantic import BaseModel, ConfigDict


class CategoryPublic(BaseModel):
  model_config = ConfigDict(from_attributes=True)

  id: int
  name: str
  slug: str
  icon_key: str
  sort_order: int


class AmenityPublic(BaseModel):
  model_config = ConfigDict(from_attributes=True)

  id: int
  name: str
  slug: str
  icon_key: str
  amenity_group: str

File: frontend/src/types/catalog.ts
/**
 * Catalog types for categories and amenities.
 */

export interface Category {
  id: number;
  name: string;
  slug: string;
  icon_key: string;
  sort_order: number;
}

export interface Amenity {
  id: number;
  name: string;
  slug: string;
  icon_key: string;
  amenity_group: string;
}
